### ISSUE TITLE
'"x" is not a known member of "None"' inside a try-except with `AttributeError`

### ISSUE THREAD

**Issue by richarddewit** [microsoft/pyright#5088]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
The problem **"x" is not a known member of "None"** inside a try-except block with `AttributeError` should not show, because it's expected

**To Reproduce**
```python
def function():
    import re

    regex = re.compile(r"^asdf-([A-Z]+)$")
    try:
        something = regex.search("123456789").group(1)
    except AttributeError:
        return None

    return something
```

**Expected behavior**
The error should not show

**Screenshots or Code**
See **To Reproduce** ⬆️ 

**VS Code extension or command-line**
Pylance v2023.5.10 (in VSCode)

**Comment by erictraut** [comment]
Pyright's behavior here is correct. There is a type violation, and pyright is reporting it. Other static type checkers like mypy generate the same error here.

You appear to be relying on exception handling for basic control flow, which is not a recommended practice. Here is a way to restructure the code to avoid a type violation error:

```python
def function():
    regex = re.compile(r"^asdf-([A-Z]+)$")
    x = regex.search("123456789")
    if x is None:
        return None
    return x.group(1)
```

**Comment by richarddewit** [comment]
> You appear to be relying on exception handling for basic control flow, which is not a recommended practice

🤨 You never heard of EAFP? It is idiomatic even:
https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/

Guess I'll put `reportOptionalMemberAccess` to `false` then.
